tests/test_freshdesk_parent_child_sync.py

```diff
@@ -0,0 +1,128 @@
+                    "updated_at": minimum_bookmark
+                },
+                "tickets_deleted": {
+                    "updated_at": minimum_bookmark
+                },
+                "tickets_spam": {
+                    "updated_at": minimum_bookmark
+                },
+                "conversations": {
+                    "updated_at": maximum_bookmark
+                }
+            }
+        }
+        stream_to_test = {'tickets'}
+        self.run_test(new_state, stream_to_test,
+                      minimum_bookmark, maximum_bookmark)
+
+    def run_test(self, new_state, stream_to_test, minimum_bookmark, maximum_bookmark):
+        """
+        Test case to verify the working of parent-child streams
+        Prerequisite:
+            - Set child bookmark is earlier than parent bookmark
+            - Set Parent bookmark is earlier than child bookmark
+            
+        • Verify that minimum bookmark is used for selected parent-child stream.
+        • Verify that records between the bookmark values are replicated.
+        """
+
+        # To collect "time_entries", "satisfaction_ratings"(child streams of "tickets") pro account is needed.
+        # Skipping them for now.
+        expected_streams = {'tickets', 'conversations'}
+
+        expected_replication_keys = self.expected_replication_keys()
+
+        conn_id = connections.ensure_connection(self)
+
+        menagerie.set_state(conn_id, new_state)
+
+        # Run in check mode
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # Table and field selection
+        catalog_entries = [catalog for catalog in found_catalogs
+                           if catalog.get('tap_stream_id') in expected_streams]
+
+        self.perform_and_verify_table_and_field_selection(
+            conn_id, catalog_entries)
+
+        # Run a sync job using orchestrator
+        sync_record_count = self.run_and_verify_sync(conn_id)
+        sync_records = runner.get_records_from_target_output()
+
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+
+                # Collect information for assertions
+                sync_messages = [record.get('data') for record in
+                                 sync_records.get(stream, {'messages': []}).get('messages')
+                                 if record.get('action') == 'upsert']
+
+                replication_key = list(expected_replication_keys[stream])[0]
+                sync_start_date_ts = self.dt_to_ts(new_state.get("bookmarks", {stream: None}).get(
+                    stream).get(replication_key), self.BOOKMARK_FORMAT)
+
+                for record in sync_messages:
+                    # Verify that the minimum bookmark is used for selected parent-child stream.
+                    replication_key_value = self.dt_to_ts(
+                        record.get(replication_key), self.BOOKMARK_FORMAT)
+
+                    self.assertGreaterEqual(
```

Comment:
Add comment

Reply:
Added comment for the following assertion.